spark3/src/test/java/org/apache/iceberg/spark/sql/TestSelect.java

```diff
@@ -63,11 +63,25 @@ public void removeTables() {
 
   @Test
   public void testSelect() {
-    List<Object[]> expected = ImmutableList.of(row(1L, "a"), row(2L, "b"), row(3L, "c"));
+    List<Object[]> expected = ImmutableList.of(
+        row(1L, "a", 1.0F), row(2L, "b", 2.0F), row(3L, "c", Float.NaN));
 
     assertEquals("Should return all expected rows", expected, sql("SELECT * FROM %s", tableName));
   }
 
+  @Test
+  public void testSelectRewrite() {
+    List<Object[]> expected = ImmutableList.of(row(3L, "c", Float.NaN));
+
+    assertEquals("Should return all expected rows", expected,
+        sql("SELECT * FROM %s where float = float('NaN')", tableName));
+
+    Assert.assertEquals("Should create only one scan", 1, scanEventCount);
+    Assert.assertEquals("Should push down expected filter",
+        "(float IS NOT NULL AND float = NaN)",
```

Comment:
Shouldn't this be `is_nan(float)` instead of `= NaN`?

Reply:
This is because in `DescribeExpressionVisitor ` we translate is_nan to `= NaN` in [here](https://github.com/apache/iceberg/blob/master/spark3/src/main/java/org/apache/iceberg/spark/Spark3Util.java#L536). Do you want me to change this to `is_nan(float)`?

Reply:
Yes, I think so. The description shouldn't produce a predicate that we don't support!